core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -843,7 +843,13 @@ private[spark] class MapOutputTrackerWorker(conf: SparkConf) extends MapOutputTr
         if (fetchedStatuses == null) {
           logInfo("Doing the fetch; tracker endpoint = " + trackerEndpoint)
           val fetchedBytes = askTracker[Array[Byte]](GetMapOutputStatuses(shuffleId))
-          fetchedStatuses = MapOutputTracker.deserializeMapStatuses(fetchedBytes, conf)
+          try {
+            fetchedStatuses = MapOutputTracker.deserializeMapStatuses(fetchedBytes, conf)
+          } catch {
+            case e: SparkException =>
```

Comment:
The failure could be `DIRECT`, how can you ensure it's only catching exception from broadcast?

Reply:
Oh, never mind. I saw the code in the following.